bf2fc6cc711aee1a0c2a/kas-fleet-manager
# feat: accept billing_model kafka create param

## Description

accept an additional optional kafka create param: `billing_model`. Inferred if not set. If provided, the param forces the AMS Quota Cost used for billing to match the requested type (either marketplace for consumption based billing or standard for pre-paid).

## Verification Steps

(use the linked cloud account id in our org)

1. Run KFM with `quota_type` set to `ams` against a fake cluster
2. Create a Kafka instance with `billing_model` set to `standard` and provide `billing_cloud_account_id` -> should not work
3. Create a Kafka instance with `billing_model` set to `marketplace` and provide a `billing_cloud_account_id` -> should work
4. Create a Kafka instance with `billing_model` set to `standard` and don't provide `billing_cloud_account_id` -> should work
5. Create a Kafka instance with no `billing_model` and `billing_cloud_account_id`  not set -> should work, `standard` inferred
6. Create a Kafka instance with no `billing_model` and `billing_cloud_account_id`  set -> should work, `marketplace` inferred

Make sure that the `billing_model` is returned to the UI in the Kafka GET and LIST endpoints.
